rtneuron/rtneuron/data/Neuron.h

```diff
@@ -69,7 +69,7 @@ class Neuron
        @param gid The gid of the neuron
        @param circuit The circuit from this this neuron comes from
     */
-    Neuron(uint32_t gid, const CircuitPtr& circuit);
+    Neuron(uint32_t gid, const CircuitCachePtr& circuit);
```

Comment:
typically there's no reason to pass a shared_ptr be passed by const ref because the use cases are:
a) store a copy of the pointer -> explicitly pass it by value
b) not interested in the pointer -> pass the object by const ref

Reply:
I remember having discussed this in the past. Performance could be a reason, and not having to worry about std::move the pointer in the function body for that purpose. Anyway, it's hard to tell how noticeable the difference is because that depends on many other factors.